File: src/ruby/Fila.rb

class Queue
    def initialize()
        @queue = []
    end
    
    def enqueue(value)
        @queue.push(value)
    end
    
    def dequeue()
        @queue.delete_at(0)
    end
    
    def display()
        print @queue.to_s + "\n"
    end
end

q = Queue.new()

q.enqueue(1)
q.enqueue(2)
q.enqueue(3)
q.enqueue(4)
q.display()

q.dequeue()
q.dequeue()
q.display()

File: src/swift/fila.swift
//
//  fila.swift
//
//  Created by Matheus Torres on 14/11/20.
//

class Queue {
    var queue: [Int] = []
    
    func enqueue(_ value: Int) {
        self.queue.append(value)
    }
    
    func dequeue(at indexPath: Int) {
        self.queue.remove(at: indexPath)
    }
    
    func show() {
        print(self.queue)
    }
    
    func clear() {
        self.queue = []
    }
}

let queue = Queue()

for n in 0...10 { queue.enqueue(n) }
queue.show()
queue.dequeue(at: 2)
queue.dequeue(at: 6)
queue.show()





